src/Orleans/SystemTargetInterfaces/IMultiClusterGossipService.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Orleans.MultiCluster;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Orleans.Runtime
+{
+    internal interface IMultiClusterGossipService : ISystemTarget
+    {
+        /// <summary>
+        /// One-way small-scale gossip: send partial data to recipient
+        /// </summary>
+        /// <param name="data">The gossip data</param>
+        /// <param name="forwardLocally">Whether to forward the changes to local silos</param>
+        /// <returns></returns>
+        Task Push(IMultiClusterGossipData gossipData, bool forwardLocally);
+
+        /// <summary>
+        /// Two-way bulk gossip: send all known data to recipient, and receive all unknown data
+        /// </summary>
+        /// <param name="gossipData">The pushed gossip data</param>
+        /// <returns>The returned gossip data</returns>
+        Task<IMultiClusterGossipData> PushAndPull(IMultiClusterGossipData gossipData);
+
+        /// <summary>
+        /// Find silos whose configuration does not match the expected configuration.
+        /// </summary>
+        /// <param name="expected">the configuration to compare with</param>
+        /// <param name="forwardLocally">whether to recursively include all silos in the same cluster</param>
+        /// <returns></returns>
+        Task<Dictionary<SiloAddress,MultiClusterConfiguration>> FindUnstableSilos(MultiClusterConfiguration expected, bool forwardLocally);
```

Comment:
I like the name. And I agree that returning just the silos is probably enough. I have updated the code.